app/Http/Controllers/UserController.php

```diff
@@ -34,6 +37,79 @@ public function performLogoutUser(Request $request) {
         return redirect()->route('index');
     }
 
+    public function performOpenIDConnect(Request $request) {
```

Comment:
@Jamie0 I would suggest a more agnostic approach here. For example, I would add a factory and strategy pattern here to determine the authentication system based on the configuration. 

I would suggest adding an additional env var to determine which Authentication system to use. Ex, `default`, `openid`, `oauth2.0`. This way, you can check which authentication system is configured and then instantiate the respective strategy for authenticating.